pkg/ingester/stream.go

```diff
@@ -310,7 +310,10 @@ func (s *stream) Push(
 				}
 			}()
 		}
+	}
 
+	if len(s.chunks) != prevNumChunks {
+		memoryChunks.Add(float64(len(s.chunks) - prevNumChunks))
 	}
```

Comment:
This code should be executed before the failure handling because some chunks may be stored in memory even if others is failed

Reply:
Great catch!